homeassistant/helpers/config_validation.py

```diff
@@ -665,6 +665,9 @@ def template(value: Any | None) -> template_helper.Template:
 
     template_value = template_helper.Template(str(value), async_get_hass_or_none())
 
+    if isinstance(value, template_helper.RenderedTemplateResultStrWrapper):
+        template_value.is_static = True
+
```

Comment:
Simply flagging the `Template`s generated by `config_validation` functions with `is_static=True` avoids having to change any type expectations by consumers of data processed through the `Schema` validation.